R/testGeno.R

```diff
@@ -102,31 +79,34 @@ testGenoSingleVar <- function(nullmod, G, E = NULL, test = c("Score", "Score.SPA
 
 
 
-.testGenoSingleVarScore <- function(Gtilde, G, resid){
+.testGenoSingleVarScore <- function(Gtilde, G, resid, RSS0){
     GPG <- colSums(Gtilde^2) # vector of G^T P G (for each SNP)
+    score.SE <- sqrt(GPG)
     score <- as.vector(crossprod(G, resid)) # G^T P Y
-    Stat <- score/sqrt(GPG)
+    Stat <- score/score.SE
     
-    res <- data.frame(Score = score, Score.SE = sqrt(GPG), Score.Stat = Stat, 
-                      Score.pval = pchisq(Stat^2, df = 1, lower.tail = FALSE) )
+    res <- data.frame(Score = score, Score.SE = score.SE, Score.Stat = Stat, 
+                      Score.pval = pchisq(Stat^2, df = 1, lower.tail = FALSE),
+                      Est.approx = score/GPG, Est.SE.appox = 1/score.SE, 
+                      PVE.approx = (Stat^2)/RSS0 # RSS0 = (n-k) when gaussian; not when binary
     
     return(res)
 }
 
 
 
-.testGenoSingleVarWald <- function(Gtilde, Ytilde, n, k){
-    GPG <- colSums(Gtilde^2) # vector of G^T P G (for each SNP)
-    GPY <- as.vector(crossprod(Gtilde, Ytilde)) # vector of G^T P Y (for each SNP)
-    beta <- GPY/GPG
-    sY2 <- sum(Ytilde^2)
-    RSS <- as.numeric((sY2 - GPY * beta)/(n - k - 1))
-    Vbeta <- RSS/GPG
-    Stat <- beta/sqrt(Vbeta)
-    res <- data.frame(Est = beta, Est.SE = sqrt(Vbeta), Wald.Stat = Stat, 
-                      Wald.pval = pchisq(Stat^2, df = 1, lower.tail = FALSE))
-    return(res)
-}
+# .testGenoSingleVarWald <- function(Gtilde, Ytilde, n, k){
+#     GPG <- colSums(Gtilde^2) # vector of G^T P G (for each SNP)
+#     GPY <- as.vector(crossprod(Gtilde, Ytilde)) # vector of G^T P Y (for each SNP)
+#     beta <- GPY/GPG
+#     sY2 <- sum(Ytilde^2)
+#     RSS <- as.numeric((sY2 - GPY * beta)/(n - k - 1))
+#     Vbeta <- RSS/GPG
+#     Stat <- beta/sqrt(Vbeta)
+#     res <- data.frame(Est = beta, Est.SE = sqrt(Vbeta), Wald.Stat = Stat, 
+#                       Wald.pval = pchisq(Stat^2, df = 1, lower.tail = FALSE))
+#     return(res)
+# }
 
 
 .testGenoSingleVarWaldGxE <- function(nullmod, G, E, GxE.return.cov.mat = FALSE){
```

Comment:
@smgogarten GxE is still "Wald"...it's a bit trickier to change this. I think I probably could, but I'm not sure if we should.

Reply:
I think it's ok to leave it.